addon/-private/system/store.js

```diff
@@ -2017,9 +2017,15 @@ Store = Service.extend({
     heimdall.increment(_load);
     let internalModel = this._internalModelForId(data.type, data.id);
 
+    let isUpdate = internalModel.currentState.isEmpty === false;
+
     internalModel.setupData(data);
 
-    this.recordArrayManager.recordDidChange(internalModel);
+    if (isUpdate) {
+      this.recordArrayManager.recordDidChange(internalModel);
```

Comment:
previously, when pushing a new record this was firing, even though a new record has no updates. It merely starts off its existence in that state...

This resulted in every new record going through the record-array manager twice on initial push...